WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java

```diff
@@ -776,6 +870,26 @@ private void showPluginRemoveFailedSnackbar() {
                 .show();
     }
 
+    private void showDomainCreditsCheckProgressDialog() {
+        if (mCheckingDomainCreditsProgressDialog == null) {
+            mCheckingDomainCreditsProgressDialog = new ProgressDialog(this);
+            mCheckingDomainCreditsProgressDialog.setCancelable(false);
```

Comment:
I think setting cancelable to false can result in an infinite dialog since the `OnPlansFetched` event may get lost during a configuration change. 
I think this isn't handled in many places in the app, but I just wanted to mention it. We may want to consider either allowing the user to cancel the dialog or setting @Subscribe(sticky=true,..) and handle it appropriately. Wdyt?

Reply:
I don't believe we can use sticky events with current Dispatcher implementation, as it only posts normal ones. I changed the dialog to be cancelable, and we will ignore results of domain credits check if they arrive when the dialog was canceled (I assume user moved on to something else)

Reply:
> I changed the dialog to be cancelable, and we will ignore results of domain credits check if they arrive when the dialog was canceled (I assume user moved on to something else)

I completely agree with this solution, thanks! ;)

>I don't believe we can use sticky events with current Dispatcher implementation, as it only posts normal ones.

Just out of curiosity, what do you mean by `it only posts normal ones`? We use sticky events on several places in the app eg. MediaBrowserActivity, PublicizeListFragment ..

Reply:
Maybe I'm missing something, but in my understanding, to receive sticky events you need to post them as sticky, using `postSticky()` method of EventBus. `Dispatcher` implementation in Flux-C is using `post()` method and not `postSticky()`.

I'm not really sure what is going on in `MediaBrowserActivity` and `PublicizeListFragment` to be honest. Looks like in addition to Dispatcher we are using a local instance of EventBus, I assume for local sticky events (that are not coming from Flux-C).

Reply:
Ohh, I see. My bad sorry. I completely forgot these events need to be sent as sticky. Thanks for the clarification!